crates/networking/manager/src/service.rs

```diff
@@ -106,14 +108,26 @@ impl NetworkManagerService {
         let status = beacon_chain.build_status_request().await?;
 
         let network = Network::init(executor.clone(), &network_config, status).await?;
+        let meta_data = network.network_state().meta_data.read().clone();
```

Comment:
Also here maybe just my lack of knowledge, but I'm not sure if cloning the `RwLock` directly actually maintains the original reference.

Because if it's not maintaining the reference, we're risking `NodeInfo` data being stale from ip address changes and metadata (attnets, syncnets) changes during runtime.

The [documentation for Arc](https://doc.rust-lang.org/std/sync/struct.Arc.html#cloning-references) does specifically mention that it points to the same reference, but [not mentioned for `RwLock`](https://doc.rust-lang.org/std/sync/struct.RwLock.html).

Would be nice if this is confirmed by a unit test if not by documentation.

Reply:
By default rust objects are non-mutable, RwLock makes the Arc writable, although yes what you mentioned above about the data becoming stale is true,

We shouldn't add NodeInfo and just extend NetworkState